target_decisioning_engine/tests/test_utils.py

```diff
@@ -98,3 +106,26 @@ def test_has_remote_dependency_mboxes_and_views(self):
         self.assertEqual(result.get("remote_needed"), True)
         self.assertEqual(result.get("remote_mboxes"), {"mbox1", "mbox3"})
         self.assertEqual(result.get("remote_views"), {"view1", "view3"})
+
+    def test_determine_artifact_location(self):
+        config = DecisioningConfig("MyClient", "12345@AdobeOrg", environment=ENVIRONMENT_DEV,
+                                   cdn_environment=ENVIRONMENT_STAGE)
+        artifact_location = determine_artifact_location(config)
+        self.assertEqual(artifact_location,
+                         "https://assets.staging.adobetarget.com/MyClient/development/v1/rules.json")
+
+    def test_determine_artifact_location_invalid_env(self):
+        config = DecisioningConfig("MyClient", "12345@AdobeOrg", environment="bad")
```

Comment:
IIRC, the Target UI allows customers to name the environment to anything.  So `bad` is a legitimate env name, but our SDKs are hard coded to production or stage.